src/mpiwrap/message_passing.F

```diff
@@ -1140,11 +1143,23 @@ SUBROUTINE mp_world_init(mp_comm)
             IF (ierr /= 0) CALL mp_stop(ierr, "mpi_init @ mp_world_init")
 !$       ELSE
 !$OMP MASTER
+#if defined(__DLAF)
+            ! DLA-Future requires that the MPI library supports
+            ! THREAD_MULTIPLE mode
+!$          CALL mpi_init_thread(MPI_THREAD_MULTIPLE, provided_tsl, ierr)
+#else
 !$          CALL mpi_init_thread(MPI_THREAD_SERIALIZED, provided_tsl, ierr)
+#endif
 !$          IF (ierr /= 0) CALL mp_stop(ierr, "mpi_init_thread @ mp_world_init")
+#if defined(__DLAF)
+!$          IF (provided_tsl .LT. MPI_THREAD_MULTIPLE) THEN
+!$             CALL mp_stop(0, "MPI library does not support the requested level of threading (MPI_THREAD_MULTIPLE).")
```

Comment:
Maybe mention in the message that one has the choice between `MPI_THREAD_SERIALIZED` and DLA-Future.